### PR TITLE
Add Wind+PEM tracking and stochastic bidding documentation

### PR DESCRIPTION
**Pull request by Xinhe-Chen**
## Summary/Motivation:
Add documentation for wind+pem renewable IES tracking and stochastic bidding. 

## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by Xinhe-Chen** [comment]
@adowling2 @dguittet This is ready to review.

**Comment by adowling2** [comment]
I'll review this later tonight.

**Comment by adowling2** [comment]
@radhakrishnatg  could be another good reviewer

**Comment by andrewlee94** [comment]
@adowling2 Note however that we hope to cut a release candidate today (pending a Pyomo release), so you might need to work with @ksbeattie and @lbianchi-lbl if this needs to be part of the February release.

**Comment by dguittet** [comment]
adding @bknueven also for his input

**Comment by dguittet** [comment]
I'm confused as to why this is being merged into the documentation at this point in time. Where is the actual code? There is already an existing Bidder.py in IDAES but it isn't specific for Wind + PEM. Is the Wind + PEM feature going to be a modification of the existing Bidder.py, or is there supposed to be another piece of code in addition? Is there going to be in IDAES many different Bidders for different technology use cases?

**Comment by Xinhe-Chen** [comment]
> I'm confused as to why this is being merged into the documentation at this point in time. Where is the actual code? There is already an existing Bidder.py in IDAES but it isn't specific for Wind + PEM. Is the Wind + PEM feature going to be a modification of the existing Bidder.py, or is there supposed to be another piece of code in addition? Is there going to be in IDAES many different Bidders for different technology use cases?

@dguittet I think this is what we are going to implement next stage and we will have the docs in advance.

**Comment by bknueven** [comment]
I am also a bit confused. It would seem like this documentation should *also* exist in dispatches. If IDAES were to document an example shouldn't it start with the one already included? https://github.com/IDAES/idaes-pse/tree/main/idaes/apps/grid_integration/examples.

**Comment by andrewlee94** [comment]
@Xinhe-Chen I also find it strange that you are adding documentation for something that does not exist yet. It is fine to work on it ahead of time, but it generally shouldn't be added to the actual docs until the code that goes with it is ready. Otherwise, user reading the docs will expect these features to be there, and then be confused when they are not.

**Review by andrewlee94 — CHANGES_REQUESTED** [review]

File: docs/reference_guides/apps/grid_integration/Bidder.rst
Comment (dguittet) [reply docs/reference_guides/apps/grid_integration/Bidder.rst:35]: This constraint makes sense but I'm wondering if this constraint (combined also with 17) is flexible enough to allow for the use case where the PEM is used in RT when the plant's DA bid is accepted but in RT the grid doesn't actually take the option of that DA power. Or perhaps other cases... @bknueven any thoughts?

File: docs/reference_guides/apps/grid_integration/Bidder.rst
Comment (Xinhe-Chen) [reply docs/reference_guides/apps/grid_integration/Bidder.rst:60]: In the manuscript of RE paper, we only have one f_t. IMO, in DA we use DA cf and in RT we use RT cf.

**Comment by Xinhe-Chen** [comment]
> I am also a bit confused. It would seem like this documentation should _also_ exist in dispatches. If IDAES were to document an example shouldn't it start with the one already included? https://github.com/IDAES/idaes-pse/tree/main/idaes/apps/grid_integration/examples.

I think we need this PR for a milestone regarding hydrogen co-production.

**Comment by Xinhe-Chen** [comment]
Decision on Feb 16: Merge this PR after the codes are merged.

File: docs/reference_guides/apps/grid_integration/Bidder.rst
Comment (Xinhe-Chen) [reply docs/reference_guides/apps/grid_integration/Bidder.rst:35]: Yes, eqn 7 has an typo. I will fix that. 

I think we may increase the weight of the hydrogen revenue at the DA bidding so that the electricity bid will be more conservative. In such case, we can reduce the underbidding. But the cost is we will lose some revenue from DA market.

File: docs/reference_guides/apps/grid_integration/Bidder.rst
Comment (Xinhe-Chen) [reply docs/reference_guides/apps/grid_integration/Bidder.rst:32]: P_{t,s}^{wind} is bounded by the f_{t}P_{wind}^{max}. Here we can use DA cf in DA problem and RT cf in RT problem.

**Comment by ksbeattie** [comment]
I've converted this to a draft PR since it appears that this won't be part of the Feb (2.0) release.  Please let me know if that's not going to be the case.

**Comment by ksbeattie** [comment]
@Xinhe-Chen, @dguittet, @adowling2 is this still going to be done (in idaes)?

**Comment by adowling2** [comment]
Yes.

**Comment by ksbeattie** [comment]
> Yes.

I've added this to the Nov release board and updated the branch since it's been quite a while.  Let me know if Nov is not a practical timeline for this.

**Comment by ksbeattie** [comment]
@adowling2, the Nov release is coming up.  Is this going to happen in the next few weeks?

**Comment by adowling2** [comment]
No, push to Feb.

**Comment by ksbeattie** [comment]
@Xinhe-Chen & @adowling2 can this be updated, reviewed and merge for *this* Feb 2024 release?

**Comment by Xinhe-Chen** [comment]
@ksbeattie @adowling2 Sorry for the late reply. The wind + PEM bidder has been done in dispatches. I will clean up the codes and add tests to merge them into IDAES.
I don't think I can do this in the Feb release, I would like to postpone this PR to the March release. Many thanks.

**Comment by ksbeattie** [comment]
> @ksbeattie @adowling2 Sorry for the late reply. The wind + PEM bidder has been done in dispatches. I will clean up the codes and add tests to merge them into IDAES. I don't think I can do this in the Feb release, I would like to postpone this PR to the March release. Many thanks.

I've taken this off the Feb release board.  When we create the next release board, please come to the dev call to have this added back to that board.

**Comment by ksbeattie** [comment]
@adowling2 , @Xinhe-Chen: Might this now be a candidate for a May release?

**Comment by Xinhe-Chen** [comment]
> @adowling2 , @Xinhe-Chen: Might this now be a candidate for a May release?

Yes. I am still working on this PR and it will be good to put it on May release.

**Comment by Xinhe-Chen** [comment]
The documentation is moved to https://github.com/IDAES/idaes-pse/pull/1407

**Comment by andrewlee94** [comment]
@Xinhe-Chen Seeing as the documentation has been moved, can we close this PR?

**Comment by Xinhe-Chen** [comment]
> @Xinhe-Chen Seeing as the documentation has been moved, can we close this PR?

Yes I think we can close this PR.

**Pull request closed without merging**
